### PR TITLE
[Finder] no PHP warning on empty directory iteration

### PR DESCRIPTION
**Pull request by ggottwald**
| Q | A |
| --- | --- |
| Branch? | 2.8 |
| Bug fix? | yes |
| New feature? | no |
| BC breaks? | no |
| Deprecations? | no |
| Tests pass? | yes |
| Fixed tickets |  |
| License | MIT |
| Doc PR | Since RecursiveDirectoryIterator::SKIP_DOTS is set as flag, opendir gets a warning if an empty directory is reached |

### CODE REVIEW

File: src/Symfony/Component/Finder/Iterator/RecursiveDirectoryIterator.php [hunk 1]
```diff
@@ -137,6 +137,10 @@ public function isRewindable()
             return $this->rewindable;
         }
 
+        if ($this->getPath() === '') {
+            return $this->rewindable = false;
+        }
```
Comment (jameshalsall) [inline src/Symfony/Component/Finder/Iterator/RecursiveDirectoryIterator.php]: A test should probably be added to cover this scenario

**Comment by fabpot** [comment]
@ggottwald Can you update the empty title to something relevant? Thanks.

**Comment by ggottwald** [comment]
https://gist.github.com/ggottwald/07194dc7926ae9093b773ec2297116f3
The changes will fix a warning with HHVM opening an empty directory. Any other ideas resolving this issue? I will also open an issue at HHVM.

**Comment by ggottwald** [comment]
https://github.com/facebook/hhvm/issues/7281

**Comment by nicolas-grekas** [comment]
@ggottwald could you please add a test case?

**Comment by ggottwald** [comment]
Sorry for the late response. Until now I was not able to force a catchable warning in the test.
In production I got a errorhandler catching the warning and throws an exception. I didn't find a way to reproduce this in the unittest.
This fix would also only be a workaround of a HHVM bug. Maybe we wait until it's fixed on their side.

**Comment by nicolas-grekas** [comment]
:+1:
Status: reviewed

File: src/Symfony/Component/Finder/Iterator/RecursiveDirectoryIterator.php [hunk 2]
```diff
@@ -137,6 +137,10 @@ public function isRewindable()
             return $this->rewindable;
         }
 
+        if ($this->getPath() === '') {
```
Comment (fabpot) [inline src/Symfony/Component/Finder/Iterator/RecursiveDirectoryIterator.php]: should be using a Yoda condition: `'' === $this->getPath()`

**Comment by fabpot** [comment]
Thank you @ggottwald.

**Pull request closed without merging**